### ISSUE TITLE
Plugin is not compatible with H2 and GraalVM JDK11

### ISSUE THREAD

**Issue by ilopmar** [micronaut-projects/micronaut-gradle-plugin#111]
It seems there is some kind of incompatibility with the plugin and GraalVM when using H2 and JDK11 (it works ok with JDK8).

I've upgraded all the test applications to use the gradle plugin and all the ones using H2 and JDK11 fail with the same error when building the native image. I honestly don't know what is going on but the same applications for other databases work properly.

To reproduce the issue:

- `git clone https://github.com/micronaut-graal-tests/micronaut-data-jdbc-graal`
- `cd micronaut-data-jdbc-graal`
- `git checkout h2-issue-jdk11`
- `sdk use java 21.0.0.r11-grl`
- `./gradlew nativeImage`

This is the diff with the upgrade to use the Gradle plugin: https://github.com/micronaut-graal-tests/micronaut-data-jdbc-graal/commit/d07d5628a8b55ac92ae3d8cde253bf8002e6b264

The only difference I've noticed is that now Launch puts `micronaut-jdbc-hikari` in `implementation` scope instead of `runtimeOnly`. In any case I've tried with this change and it still fails.

If you checkout the commit before doing that change (`git checkout f2707fd2a1b4d74fd7157bd1596d4f0da3170591`) and try to generate the native image again (running `./build-native-image.sh`), everything works as expected.

As mentioned, this also happens in the test applications for Data-JPA, Flyway and Liquibase and only when using H2 (and JDK11).

**Comment by graemerocher** [comment]
I don't see how this is related to the plugin

**Comment by ilopmar** [comment]
Me neither but after the upgrade with that commit it fails. And it happens in more than one app... :shrug:

**Comment by graemerocher** [comment]
For me this fails with the same error on JDK 11 with 20.3 and Micronaut 2.2.x so there doesn't appear to be any regression here

**Comment by ilopmar** [comment]
Yes, but that only happens when using the plugin. If you checkout the commit I mentioned (which is the previous one before introducing the plugin in the test app) it works.

**Comment by graemerocher** [comment]
Right yes, just confirming this is not a new regression

**Comment by ilopmar** [comment]
Yes, you are right. It's been there for a long time, but as we haven't used the plugin before for those apps, we haven't noticed.

**Comment by ilopmar** [comment]
I'm going to rollback the use of the plugin in the tests applications affected by this because I want to make sure they work and we are not hiding any other issue. 

I've also updated the description of this issue to use a new branch `h2-issue-jdk11`.

**Comment by ilopmar** [comment]
This has been fixed in GraalVM 21.1 (currently `master` branch). When I did the rollback of the plugin in the tests applications I missed the Liquibase one and this morning, while testing other stuff I've seen that it works with the plugin.
As this is failing in GraalVM 21.0 I'll keep the Liquibase-H2-JDK11 job enabled only for the GraalVM-dev branch. Once 21.1 is released I'll upgrade the other tests applications to use the plugin.
